Rakefile

```diff
@@ -6,41 +6,48 @@ require 'bigdecimal'
 require 'shellwords'
 require 'rake/testtask'
 autoload :Xgt, 'xgt/ruby'
+autoload :Etc, 'etc'
 
 def mining_disabled?
   ENV['MINING_DISABLED']&.upcase == 'TRUE'
 end
 
 def mining_threads
-  ENV['MINING_THREADS'] || 1
+  ENV['MINING_THREADS'] || Etc.nprocessors
```

Comment:
You can avoid this, just set it to 0 or omit the config option and it'll spawn a thread per core.